### PR TITLE
feat(cb2-7397): add the small trailer vehicle type

### PR DESCRIPTION
**Pull request by tahayazici**
## View a Small Trailer Tech Record

[CB2-7397](https://dvsa.atlassian.net/browse/CB2-7397)

### CODE REVIEW

File: src/domain/VehicleFactory.ts [hunk 1]
```diff
@@ -33,6 +34,8 @@ export class VehicleFactory {
         return new processors.LgvProcessor(vehicleObj as LightGoodsVehicle, techRecordDAO);
       case enums.VEHICLE_TYPE.CAR:
         return new processors.CarProcessor(vehicleObj as Car, techRecordDAO);
+      case enums.VEHICLE_TYPE.SMALL_TRL:
+        return new processors.SmallTrailerProcessor(vehicleObj as SmallTrailer, techRecordDAO);
```
Comment (aaron-downing) [inline src/domain/VehicleFactory.ts:38]: I've been trying to rack my brains for a clean way we can remove the type casting `vehicleObj as ...` but I think unless we use chained if statements that utilise type guards, there's no other way....

```
function isSmallTrl(vehicleObj: Vehicle): vehicleObj is SmallTrailer {
    return vehicleObj.techRecord[0].vehicleType === enums.VEHICLE_TYPE.SMALL_TRL
}

if (isSmallTrl(vehicleObj)) {
  return new processors.SmallTrailerProcessor(vehicleObj, techRecordDAO);
}
else if {...}
```

File: src/domain/Processors/SmallTrailerProcessor.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  protected validateTechRecordFields(newVehicle: CarLgvTechRecord): string[] {
+    console.log("Small trailer validate tech record fields");
+
+    const validationResult = validators.smallTrailerValidation.validate(newVehicle, this.validationOptions);
+
+    return validators.handleValidationResult(validationResult);
+  }
+
+  protected mapFields(techRecord: CarLgvTechRecord): CarLgvTechRecord {
+    console.log(`Small trailer populate fields`);
```
Comment (aaron-downing) [inline src/domain/Processors/SmallTrailerProcessor.ts:25]: Are similar `console.log` messages in other processors?

Comment (tahayazici) [reply src/domain/Processors/SmallTrailerProcessor.ts:25]: Yes. That is why I followed suit.

File: src/domain/VehicleFactory.ts (hunk 1 above)
Comment (tahayazici) [reply src/domain/VehicleFactory.ts:38]: I know typecasting is not the best approach but your suggestion introduces 7 new methods just to be used in `if else` statements. Thus, I will leave it as is.

Maybe one day we will have proper interfaces for each vehicle type. As that is the proper fix to the root problem here.

**Review by gjulien-bjss — APPROVED** [review]

**Review by Ruairidh-BJSS — APPROVED** [review]

**Review by kpearmanbjss — APPROVED** [review]

**Pull request merged**